vehicle/tesla/controller.go

```diff
@@ -13,14 +15,26 @@ const ProxyBaseUrl = "https://tesla.evcc.io"
 
 type Controller struct {
 	vehicle *tesla.Vehicle
+	current int64
+	dataG   provider.Cacheable[float64]
 }
 
 // NewController creates a vehicle current and charge controller
-func NewController(vehicle *tesla.Vehicle) *Controller {
-	impl := &Controller{
-		vehicle: vehicle,
+func NewController(ro, rw *tesla.Vehicle) *Controller {
+	v := &Controller{
+		vehicle: rw,
 	}
-	return impl
+
+	v.dataG = provider.ResettableCached(func() (float64, error) {
+		if v.current >= 6 {
+			// assume match above 6A to save API requests
+			return float64(v.current), nil
+		}
+		res, err := ro.Data()
+		return float64(res.Response.ChargeState.ChargeRate), apiError(err)
```

Comment:
ChargeRate is not the correct property to read as my test shows (see https://github.com/evcc-io/evcc/issues/13007#issuecomment-2016762205).
You should probably use "charge_amps" or "charger_actual_current".